src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -2281,46 +2283,69 @@ def create_tunnel(cmd, resource_group_name, name, port=None, slot=None):
     tunnel_server = TunnelServer('', port, host_name, profile_user_name, profile_user_password)
     _ping_scm_site(cmd, resource_group_name, name)
 
+    _wait_for_webapp(tunnel_server)
+
     t = threading.Thread(target=_start_tunnel, args=(tunnel_server,))
     t.daemon = True
     t.start()
 
-    _wait_for_tunnel(tunnel_server, False)
-    logger.warning("SSH is available ( username: %s, password: %s )", ssh_user_name, ssh_user_password)
-
-    s = threading.Thread(target=_start_ssh,
-                         args=('localhost', tunnel_server.get_port(), ssh_user_name))
+    s = threading.Thread(target=_start_ssh_session,
+                         args=('localhost', tunnel_server.get_port(), ssh_user_name, ssh_user_password))
     s.daemon = True
     s.start()
 
     while s.isAlive() and t.isAlive():
         time.sleep(5)
 
 
-def _wait_for_tunnel(tunnel_server, print_warnings):
-    if not _check_for_ready_tunnel(tunnel_server):
-        if print_warnings:
-            logger.warning('Tunnel is not ready yet, please wait (may take up to 1 minute)')
-        while True:
-            time.sleep(1)
-            if print_warnings:
-                logger.warning('.')
-            if _check_for_ready_tunnel(tunnel_server):
-                break
+def _wait_for_webapp(tunnel_server):
+    tries = 0
+    while True:
+        if is_webapp_up(tunnel_server):
+            break
+        if tries == 0:
+            logger.warning('Connection is not ready yet, please wait')
+        if tries == 60:
+            raise CLIError("Timeout Error, Unable to establish a connection")
+        tries = tries + 1
+        logger.warning('.')
+        time.sleep(1)
 
 
 def _start_tunnel(tunnel_server):
-    _wait_for_tunnel(tunnel_server, True)
     tunnel_server.start_server()
 
 
-def _start_ssh(host_name, port, user_name):
-    subprocess.call("ssh -o StrictHostKeyChecking=no {}@{} -p {}".format(user_name, host_name, port), shell=True)
+def _start_ssh_session(hostname, port, username, password):
+    tries = 0
+    while True:
+        try:
+            c = Connection(host=hostname,
+                           port=port,
+                           user=username,
+                           # connect_timeout=60*10,
+                           connect_kwargs={"password": password})
+            break
+        except Exception:  # pylint: disable=broad-except
+            if tries == 0:
```

Comment:
I would suggest you log an verbose entry to document the exception details: `logger.info(ex)`. My concern is catching  generic exceptions above could swallow non connection related errors which ends up making the diagnosis pretty hard.

Reply:
added